public/app/plugins/datasource/prometheus/metric_find_query.ts

```diff
@@ -141,22 +141,35 @@ export default class PrometheusMetricFindQuery {
     const instantQuery: PromQueryRequest = { expr: query } as PromQueryRequest;
     return this.datasource.performInstantQuery(instantQuery, end).pipe(
       map((result) => {
-        return _map(result.data.data.result, (metricData) => {
-          let text = metricData.metric.__name__ || '';
-          delete metricData.metric.__name__;
-          text +=
-            '{' +
-            _map(metricData.metric, (v, k) => {
-              return k + '="' + v + '"';
-            }).join(',') +
-            '}';
-          text += ' ' + metricData.value[1] + ' ' + metricData.value[0] * 1000;
-
-          return {
-            text: text,
-            expandable: true,
-          };
-        });
+        switch (result.data.data.resultType) {
+          case 'scalar': // [ <unix_time>, "<scalar_value>" ]
+          case 'string': // [ <unix_time>, "<string_value>" ]
+            return [
+              {
+                text: result.data.data.result[1] || '',
+                expandable: false,
+              },
+            ];
+          case 'vector':
+            return _map(result.data.data.result, (metricData) => {
+              let text = metricData.metric.__name__ || '';
+              delete metricData.metric.__name__;
+              text +=
+                '{' +
+                _map(metricData.metric, (v, k) => {
+                  return k + '="' + v + '"';
+                }).join(',') +
+                '}';
+              text += ' ' + metricData.value[1] + ' ' + metricData.value[0] * 1000;
+
+              return {
+                text: text,
+                expandable: true,
+              };
+            });
+          default:
+            throw Error(`Unknown/Unhandled result type: [${result.data.data.resultType}]`);
```

Comment:
I throw error since those are being handled and showed with a red notification. But not sure it is the best way to do it here.

Reply:
Just my opinion, but it's always best to throw an error when it does not make sense to handle it. So, this looks fine to me. With that said, what about instant vectors? Are they handled somewhere else?